Map/Source/SquareMap.cpp

```diff
@@ -62,4 +63,133 @@ SquareMap::SquareMap(const int mapSize)
     {
         _fieldList.at(i).at(0) = std::make_unique<FightField>();
     }
-}
\ No newline at end of file
+}
+
+SquareMap::SquareMap(FieldList&& fieldList) : _fieldList(std::move(fieldList))
+{
+}
+
+bool SquareMap::isFieldAccessible(const Position& position)
+{
+    if (not isField(position) or getField(position)->getType() == FieldType::Wall)
+    {
+        return false;
+    }
+    return true;
+}
+
+bool SquareMap::isField(const Position& position)
+{
+    try
+    {
+        if (getField(position))
+        {
+            return true;
+        }
+        return false;
+    }
+    catch (std::out_of_range&)
+    {
+        return false;
+    }
+}
+
+const std::unique_ptr<Field>& SquareMap::getField(const Position& position)
+{
+    return _fieldList.at(position._x).at(position._y);
+}
+
+void SquareMap::updateVisilibity(const Position& playerPosition)
+{
+    makeNonBarrierFieldsInvisible();
+    if (getField(playerPosition)->getType() == FieldType::Door)
+    {
+        makeRoomVisible(Position(playerPosition._x + 1, playerPosition._y));
+        makeRoomVisible(Position(playerPosition._x - 1, playerPosition._y));
+        makeRoomVisible(Position(playerPosition._x, playerPosition._y + 1));
+        makeRoomVisible(Position(playerPosition._x, playerPosition._y - 1));
+    }
+    else if(not isFieldAccessible(playerPosition))
+    {
+        return;
+    }
+    else
+    {
+        makeRoomVisible(playerPosition);
+    }
+}
+
+void SquareMap::makeNonBarrierFieldsInvisible()
+{
+    for(const auto& column : _fieldList)
+    {
+        for(const auto& field : column)
+        {
+            if(field->getType() != FieldType::Wall and field->getType() != FieldType::Door)
+            {
+                field->makeInvisible();
+            }
+        }
+    }
+}
+
+void SquareMap::makeRoomVisible(const Position& startPosition)
+{
+    if (not isField(startPosition))
+    {
+        return;
+    }
+
+    Position tempPosition = startPosition;
+    makeRowVisible(tempPosition);
+
+    while (isFieldAccessible(tempPosition) and getField(tempPosition)->getType() != FieldType::Door)
```

Comment:
te dwa while fajnie by bylo wrzucic do osobnych metody, byloby to duzo czytelniejsze